### ISSUE TITLE
ERROR: 'NoneType' object has no attribute 'dataProvider' See log for more details

### ISSUE THREAD

**Issue by tmr-jay** [ryersongeo/qgis_location_analytics#1]
Greetings,

Had this error while attempting to follow the case study. 

> 'NoneType' object has no attribute 'dataProvider' See log for more details

Was originally using QGIS 3.0 on Win8.1 and the script would fail to add unless i changed 
`print "Path not found"`
to
`print ("Path not found")`
but even then, there appeared to be lots of library calls that would not work (presume deprecated functions?)

I then tried the other version I had installed (2.12.3) on the same Win8.1 machine and got the error message above. 

Tried another machine (win7 using ver3.0) and again, it didn't work.

Then tried it in 2.12.3 and got the error message above, and finally tried (again on the win7 machine) the version you had used when creating the case study (2.14.3) and again got the error message above.

Being new to QGIS (and seeing in searches that others report this same error in other contexts), i'm thinking its a problem that sits to the side of the supplied scripts, however I'm hoping someone can help.

Thanks.

**Comment by crinner** [comment]
Hi, I haven't used the scripts in a while but the error message sounds like there is an issue with your output layer definition. Upon starting either of the scripts, did you specify a valid output layer?

**Comment by tmr-jay** [comment]
Indeed, when running the "distance matrix network" script, i tried Save to Temporary File as the output, but also tried saving to a specified filename on my file system (tried txt, shp and null file extension) to no avail. Same error each time.

**Comment by kman411** [comment]
Hi, I'm also experiencing the same error message. Would be great to see if there are any remedies to this. The problem occurs for both the distance - network script and the distance - euclidian script.
